Marlin/Marlin_main.cpp

```diff
@@ -2335,7 +2335,7 @@ static void clean_up_after_endstop_or_probe_move() {
 
     float measured_z = NAN;
     if (!DEPLOY_PROBE()) {
-      measured_z = run_z_probe() + zprobe_zoffset;
+      measured_z = run_z_probe() + Z_PROBE_OFFSET_FROM_EXTRUDER;
```

Comment:
probing always returns the same values for bed leveling and calibration independent of z_offset changes

Reply:
What if it were common practice to set `Z_PROBE_OFFSET_FROM_EXTRUDER` to 0 and just rely on `M851` to set the `zprobe_zoffset`?

Earlier versions of Marlin simply stored the literal result of `run_z_probe`. During printing, it would add the `zprobe_zoffset` to get the mesh Z values.

The cool thing about that was that as you changed `zprobe_zoffset` during a print the nozzle would move up and down right away, and if you saved with `M500` it preserved your change.

Reply:
Here Z_PROBE_OFFSET_FROM_EXTRUDER is set so the probe moves (clearance etc.) are calculated correctly (not the case if you set it to 0); and fine tuning is done by changing zprobe_zoffset. Think of (zprobe_zoffset -  Z_PROBE_OFFSET_FROM_EXTRUDER) as an additional z_offset that only affect the homing height.

>The cool thing about that was that as you changed zprobe_zoffset...
Not if you have bed leveling disabled. So the current approach of babystepping is the right one (applies in all cases) but you lost that info on homing (except in the homing with probe case)- that's where this patch takes over now, the info is kept in all cases.